spring-integration-amqp/src/test/java/org/springframework/integration/amqp/outbound/AmqpOutboundEndpointTests.java

```diff
@@ -177,7 +177,7 @@ public void adapterWithContentType() throws Exception {
 				.setHeader(AmqpHeaders.CONTENT_TYPE, "application/json")
 				.build();
 		this.ctRequestChannel.send(message);
-		org.springframework.amqp.core.Message m = template.receive();
+		org.springframework.amqp.core.Message m = receive(template);
```

Comment:
Why `AmqpTemplate.receive(long timeoutMillis)` doesn't work for us here?

Reply:
Because it's an auto-delete queue; when we have a timeout, we create a temporary consumer; when `receive()` exits, the queue is deleted.

This test doesn't fail often; but it's failed for me a few times when using `--parallel`.

Reply:
Oh! Right. I forgot about a difference between `channel.basicGet()` `TemplateConsumer`.

Thanks!

Merging...